grimoire/core/styling/palette.py

```diff
@@ -0,0 +1,148 @@
+from enum import Enum
+
+from gdpc import Block
+
+from grimoire.core.assets.asset import Asset
+from grimoire.core.styling.blockform import BlockForm
+from grimoire.core.styling.colors import MinecraftColor
+from grimoire.core.styling.materials.dithering import DitheringPattern
+from grimoire.core.styling.materials.material import Material, MaterialParameters
+
+
+class MaterialRole(Enum):
+    primary_wall = "primary_wall"
+    secondary_wall = "secondary_wall"
+
+    primary_roof = "primary_roof"
+    secondary_roof = "secondary_roof"
+
+    primary_stone = "primary_stone"
+    secondary_stone = "secondary_stone"
+
+    primary_wood = "primary_wood"
+    secondary_wood = "secondary_wood"
+
+    pillar = "pillar"
+
+
+class _ResolutionPriority:
+    order: list[MaterialRole]
+    next_in_line: dict[MaterialRole, MaterialRole]
+
+    def __init__(
+        self, order: list[MaterialRole], next_in_line: dict[MaterialRole, MaterialRole]
+    ):
+        self.order = order
+        self.next_in_line = next_in_line
+
+    def get_next_in_line(self, role: MaterialRole) -> MaterialRole | None:
+        if role in self.next_in_line:
+            return self.next_in_line[role]
+
+        return None
+
+
+class ResolutionPriority(Enum):
```

Comment:
This is a very strange Enum. Why does it have only one member? Why is it's value a class? A dict and some functions would probably be better here...

Reply:
The idea is that we will probably have different resolution priorities for stone-heavy architecture, wood-heavy architecture, and perhaps others. (IE a palette with a stone-heavy resolution order would use stone if it doesn't find a matching block for a role) 

I think these resolution orders can probably be a closed set- hence the `Enum`- but are complicated enough to merit their own class, so I am using the internal `_ResolutionPriority` class. We can then reference the resolution orders simply via the names in the enum, probably "stone", "wood", etc

Reply:
I feel like this is a very smelly idea, perhaps we should discuss it in a call sometime?

Reply:
Sure, definitely up to discuss this